tss-esapi/src/structures/buffers/public.rs

```diff
@@ -0,0 +1,520 @@
+    /// # Details
+    /// This is required if the public algorithm is set to
+    /// [Ecc][`crate::interface_types::algorithm::PublicAlgorithm::Ecc].
+    pub const fn with_ecc_parameters(mut self, ecc_parameters: PublicEccParameters) -> Self {
+        self.ecc_parameters = Some(ecc_parameters);
+        self
+    }
+
+    /// Adds the ECC unique identifier for the [Public] structure
+    /// to the builder.
+    ///
+    /// # Details
+    /// This is required if the public algorithm is set to
+    /// [Ecc][`crate::interface_types::algorithm::PublicAlgorithm::Ecc].
+    ///
+    /// The unique identifier is a ecc point.
+    pub fn with_ecc_unique_identifier(mut self, ecc_unique_identifier: EccPoint) -> Self {
+        self.ecc_unique_identifier = Some(ecc_unique_identifier);
+        self
+    }
+
+    /// Adds the symmetric cipher parameters for the [Public] structure
+    /// to the builder.
+    ///
+    /// # Details
+    /// This is required if the public algorithm is set to
+    /// [SymCipher][`crate::interface_types::algorithm::PublicAlgorithm::SymCipher].
+    pub const fn with_symmetric_cipher_parameters(
+        mut self,
+        symmetric_cipher_parameters: SymmetricCipherParameters,
+    ) -> Self {
+        self.symmetric_cipher_parameters = Some(symmetric_cipher_parameters);
+        self
+    }
+
+    /// Adds the symmetric cipher unique identifier for the [Public] structure
+    /// to the builder.
+    ///
+    /// # Details
+    /// This is required if the public algorithm is set to
+    /// [SymCipher][`crate::interface_types::algorithm::PublicAlgorithm::SymCipher].
+    pub fn with_symmetric_cipher_unique_identifier(
+        mut self,
+        symmetric_cipher_unique_identifier: &Digest,
+    ) -> Self {
+        self.symmetric_cipher_unique_identifier = Some(symmetric_cipher_unique_identifier.clone());
+        self
+    }
+
+    /// Builds the [Public] structure.
+    ///
+    /// # Errors
+    /// Will return error if the public algorithm, object attributes or name
+    /// hashing algorithm have not been set or if the parameters and unique identifier
+    /// does not match the selected public algorithm.
+    pub fn build(self) -> Result<Public> {
+        let algorithm = self.public_algorithm.ok_or_else(|| {
+            error!("Algorithm is required and has not been set in the PublicBuilder");
+            Error::local_error(WrapperErrorKind::ParamsMissing)
+        })?;
+
+        let object_attributes = self.object_attributes.ok_or_else(|| {
+            error!("ObjectAttributes is required and has not been set in the PublicBuilder");
+            Error::local_error(WrapperErrorKind::ParamsMissing)
+        })?;
+
+        let name_hashing_algorithm = self.name_hashing_algorithm.ok_or_else(|| {
+            error!(
+                "The name hashing algorithm is required and has not been set in the PublicBuilder"
+            );
+            Error::local_error(WrapperErrorKind::ParamsMissing)
+        })?;
```

Comment:
Really nice logging and error management

Reply:
Thank you.